src/test/scala/com/box/castle/core/committer/manager/BatchSizeManagerTest.scala

```diff
@@ -0,0 +1,121 @@
+package com.box.castle.core.committer.manager
+
+import org.specs2.mutable.Specification
+
+/**
+  * Unit Tests for BatchSizeManager
+  */
+class BatchSizeManagerTest extends Specification {
+
+
+  "track" should {
+    "track fixed number of samples only" in {
+      // Setup
+      val samplingSlots = 3
+      val samplingInterval = 100 // milliseconds
+      val bufferSize = 100 //bytes
+      val bytesRead = 10
+      val batchSizeManager = new BatchSizeManager(samplingSlots, samplingInterval, bufferSize)
+
+      // Execute
+      batchSizeManager.track(bytesRead, System.currentTimeMillis())
+      Thread.sleep(samplingInterval)
+      batchSizeManager.track(bytesRead, System.currentTimeMillis())
+      Thread.sleep(samplingInterval)
+      batchSizeManager.track(bytesRead, System.currentTimeMillis())
+      Thread.sleep(samplingInterval)
+      batchSizeManager.track(bytesRead, System.currentTimeMillis())
+      Thread.sleep(samplingInterval)
+      batchSizeManager.track(bytesRead, System.currentTimeMillis())
+
+      // Validate
+      batchSizeManager.size shouldEqual samplingSlots
+
+    }
+
+    "only keep samples occurring after specified sample interval" in {
+      // Setup
+      val samplingSlots = 3
+      val samplingInterval = 1000 // milliseconds
+      val bufferSize = 100 //bytes
+      val bytesRead = 10
+      val batchSizeManager = new BatchSizeManager(samplingSlots, samplingInterval, bufferSize)
+
+      // Execute
+      batchSizeManager.track(bytesRead, System.currentTimeMillis())
+      batchSizeManager.track(bytesRead, System.currentTimeMillis()) // This one should be discarded
+
+      // Validate
+      batchSizeManager.size shouldEqual 1
+
+    }
+  }
+
+  "getDelay" should {
+    "compute delay based on tracked samples" in {
+      // Setup
+      val samplingSlots = 3
+      val samplingInterval = 100 // milliseconds
+      val bufferSize = 100 //bytes
+      val bytesRead = 10
+      val batchSizeManager = new BatchSizeManager(samplingSlots, samplingInterval, bufferSize)
+
+      // Execute
+      val ts1 = System.currentTimeMillis()
+      batchSizeManager.track(bytesRead, ts1)
+      Thread.sleep(samplingInterval)
```

Comment:
these tests will be super flaky, you cannot Thread.sleep() here.  mock the call to System.currentTimeMillis(), or... my personal favorite is just create your own method for getting time in your BatchSizeManager and just mock that method with the value you want it to return